mozilla_django_oidc/auth.py

```diff
@@ -126,7 +126,9 @@ def authenticate(self, **kwargs):
         response = requests.post(self.OIDC_OP_TOKEN_ENDPOINT,
                                  data=token_payload,
                                  verify=import_from_settings('OIDC_VERIFY_SSL', True))
-        response.raise_for_status()
+
+        if not response.status_code == requests.codes.ok:
+            return None
```

Comment:
I'm on the fence about this. What are the situations in which we get a non-ok status code?

* user is denied -- failing the login is ok here
* service is down -- failing the login is ok here, but as a user, I'd really like some indication that the problem wasn't my fault and what I should do about it (probably wait 10 minutes and try again)
* configuration is wrong? -- failing the login is ok here, too, but as a user, I'd really like some indication that the problem wasn't my fault and what I should do about it (probably write up a bug)

Are there other scenarios where this could return a non-ok code?

Are there ways we can get a message to the user so they know what to do next?

Reply:
Returning `None` is not useful. I agree with @willkg there are many reasons why this might happen. 

Any response error code `>=500` definitely needs to bubble up. If *your* server can't talk to auth0's server it's definitely not the users' fault. Arguably it's not your fault either, but auth0's. 

A huge improvement would be something like:
```python
if response.status_code >= 500:
    raise RemoteTokenEndpointError(response.status_code)
```

Reply:
@willkg [mentioned](https://github.com/mozilla/mozilla-django-oidc/issues/154) that it's happened at least 9 times that he gets a 403 from the OIDC_OP_TOKEN_ENDPOINT. What's that about? 

It would be a small improvement just to raise a specific error just for that but I'd like to know what it means to get a 403 there there. Ideally, it should redirect to a page if possible. E.g.
```python
if response.status_code == 403:
    # That usually means ...
    if settings.import_from_settings('OIDC_403_PAGE_URL', None):
       return redirect(settings.OIDC_403_PAGE_URL)  # perhaps some use of reverse()
    else:
        raise ForbiddenOPError(response)
```